org.lflang.tests/src/org/lflang/tests/TestBase.java

```diff
@@ -555,118 +548,104 @@ static public String stackTraceToString(Throwable t) {
-        shCode.append("pids+=\"$!\"\nsleep 3\n");
-        for (Path dockerFile : dockerFiles) {
-            var composeServiceName = dockerFile.getFileName().toString().replace(".Dockerfile", "");
-            shCode.append(String.format("%s run -f %s --rm -T %s &\n",
-                dockerComposeCommand,
-                dockerComposeFilePath,
-                composeServiceName));
-            shCode.append("pids+=\" $!\"\n");
-        }
-        shCode.append("for p in $pids; do\n");
-        shCode.append("    if wait $p; then\n");
-        shCode.append("        :\n");
-        shCode.append("    else\n");
-        shCode.append("        exit 1\n");
-        shCode.append("    fi\n");
-        shCode.append("done\n");
-        return shCode.toString();
-    }
+    private static Path dockeRunScript = null;
 
     /**
-     * Returns true if docker exists, false otherwise.
+     * Return the path to a bash script containing DOCKER_RUN_SCRIPT.
+     *
+     * If the script does not yet exist, it is created.
      */
-    private boolean checkDockerExists() {
-        LFCommand checkCommand = LFCommand.get("docker", List.of("info"));
-        return checkCommand.run() == 0;
+    private Path getDockerRunScript() throws TestError {
+        if (dockeRunScript != null) {
+            return dockeRunScript;
+        }
+
+        try {
+            var file = File.createTempFile("run_docker_test", "sh");
+            file.deleteOnExit();
+            file.setExecutable(true);
+            var path = file.toPath();
+            try (BufferedWriter writer = Files.newBufferedWriter(path)) {
+                writer.write(DOCKER_RUN_SCRIPT);
+            }
+            dockeRunScript = path;
+        } catch (IOException e) {
+            throw new TestError("IO Error during test preparation.", Result.TEST_EXCEPTION, e);
+        }
+
+        return dockeRunScript;
     }
 
     /**
-     * Return a list of ProcessBuilders used to test the docker option under non-federated execution.
-     * See the following for references on the instructions called:
-     * docker build: https://docs.docker.com/engine/reference/commandline/build/
-     * docker run: https://docs.docker.com/engine/reference/run/
-     * docker image: https://docs.docker.com/engine/reference/commandline/image/
-     *
-     * @param test The test to get the execution command for.
+     * Throws TestError if docker does not exist. Does nothing otherwise.
      */
-    private List<ProcessBuilder> getNonfederatedDockerExecCommand(LFTest test) {
-        if (!checkDockerExists()) {
-            System.out.println(Message.MISSING_DOCKER);
-            return List.of(new ProcessBuilder("exit", "1"));
+    private void checkDockerExists() throws TestError {
+        if (LFCommand.get("docker", List.of()) == null) {
+            throw new TestError("Executable 'docker' not found" , Result.NO_EXEC_FAIL);
+        }
+        if (LFCommand.get("docker-compose", List.of()) == null) {
```

Comment:
Hasn't the API changed to just `docker` (no `docker-compose`)?

Reply:
I think we can just remove this check...

Reply:
You need to install docker-compose separately, though (even if you use `docker compose`). This also puzzled me at first, but they shipp as separate packages

Reply:
Should we revert the removal?